lib/rules/selector-max-specificity/index.js

```diff
@@ -192,32 +195,23 @@ const rule = (primary, secondaryOptions) => {
 			// Using `.selectors` gets us each selector in the eventuality we have a comma separated set
 			for (const selector of ruleNode.selectors) {
 				for (const resolvedSelector of resolvedNestedSelector(selector, ruleNode)) {
-					try {
-						// Skip non-standard syntax selectors
-						if (!isStandardSyntaxSelector(resolvedSelector)) {
-							continue;
-						}
-
-						parseSelector(resolvedSelector, result, ruleNode, (selectorTree) => {
-							// Check if the selector specificity exceeds the allowed maximum
-							if (
-								specificity.compare(maxChildSpecificity(selectorTree), maxSpecificityArray) === 1
-							) {
-								report({
-									ruleName,
-									result,
-									node: ruleNode,
-									message: messages.expected(resolvedSelector, primary),
-									word: selector,
-								});
-							}
-						});
-					} catch {
-						result.warn('Cannot parse selector', {
-							node: ruleNode,
-							stylelintType: 'parseError',
-						});
```

Comment:
[note] Since `parseSelector()` internally uses `try ... catch`, it's unnecessary here:

https://github.com/stylelint/stylelint/blob/3633e44eb9093c392fe92d9f6a20f4772b22fe92/lib/utils/parseSelector.js#L15-L17